### ISSUE TITLE
[Bug]: SmolVLA does not update preprocessor stats with pretrained model

### ISSUE THREAD

**Issue by alfieroddan** [open-edge-platform/physical-ai-studio#484]
### Description

In Pi05's implementation if the model exists it is assumed that the stats do not need to be re-initialized as it will lose information on normalisation pre and post.

https://github.com/open-edge-platform/physical-ai-studio/blob/475a2af96e5559376647fede22bc050a65c7ece0/library/src/physicalai/policies/pi05/policy.py#L412-L447

In SmolVLA however if the model exists we do not initialise:

https://github.com/open-edge-platform/physical-ai-studio/blob/475a2af96e5559376647fede22bc050a65c7ece0/library/src/physicalai/policies/smolvla/policy.py#L250-L282

SmolVLA does not call `reformat_dataset_to_match_policy` and so delta timestamps aren't calculated. Similarly we have no `_update_preprocessor_stats` equivalent. We therefore have no way of updating the pre-processor stats from the pretrained to the new dataset stats.

You can see how this changes for Pi05 here:

```python
from physicalai.benchmark import LiberoBenchmark
from physicalai.data import LeRobotDataModule
from physicalai.policies import Pi05
from physicalai.train import Trainer

if __name__ == "__main__":
    # # load blank policy
    # policy = Pi05()

    # # policy stats
    # print("Blank Policy Dataset Stats:", policy._dataset_stats)
    # # None — blank policy has no training data, so no stats.

    # # Policy from pre-trained weights
    # policy = Pi05("lerobot/pi05_base")
    # print("Pi05 Base Policy Dataset Stats:", policy._dataset_stats)
    # # pi05_base uses identity normalization (mean=0, std=1) — generic base model.
    # # {
    # #   'observation.state': { shape: (32,), mean: [0.0 x32], std: [1.0 x32] },
    # #   'action':            { shape: (32,), mean: [0.0 x32], std: [1.0 x32] },
    # # }

    # # what about another policy?
    # policy = Pi05(pretrained_name_or_path="lerobot/pi05_libero_finetuned_v044")
    # print("Pi05 LIBERO finetuned Policy Dataset Stats:", policy._dataset_stats)
    # # pi05_libero_finetuned_v044 has real LIBERO data stats baked in.
    # # {
    # #   'action':               { shape: (7,),        mean: [0.063, 0.087, -0.090, 0.001, 0.006, -0.005, -0.050],
    # #                                                  std:  [0.336, 0.378,  0.445, 0.039, 0.063,  0.078,  0.999] },
    # #   'observation.state':    { shape: (8,),         mean: [-0.047, 0.034, 0.765, 2.972, -0.220, -0.126, 0.027, -0.027],
    # #                                                  std:  [ 0.105, 0.152, 0.379, 0.344,  0.907,  0.325, 0.014,  0.014] },
    # #   'observation.images.*': { shape: (3, 256, 256), mean: ~ImageNet ([0.485, 0.456, 0.406]),
    # #                                                   std:  ~ImageNet ([0.229, 0.224, 0.225]) },
    # # }

    # # what about a blank policy with a custom dataset stats?
    # dm = LeRobotDataModule(repo_id="HuggingFaceVLA/libero", num_workers=0, episodes=[0])
    # policy = Pi05()
    # trainer = Trainer(max_epochs=1)
    # dm.setup("fit")
    # trainer.datamodule = dm
    # policy._trainer = trainer
    # policy.setup("fit")
    # print("Blank Policy with custom dataset stats:", policy._dataset_stats)
    # # Stats now match the LIBERO dataset (episode 0 only).
    # # {
    # #   'observation.image':  { shape: (3, 256, 256), mean: [0.487, 0.440, 0.405], std: [0.202, 0.233, 0.223] },
    # #   'observation.image2': { shape: (3, 256, 256), mean: [0.514, 0.456, 0.421], std: [0.231, 0.264, 0.250] },
    # #   'observation.state':  { shape: (8,),          mean: [-0.047, 0.034, 0.765, 2.972, -0.220, -0.126, 0.027, -0.027],
    # #                                                  std:  [ 0.105, 0.152, 0.379, 0.344,  0.907,  0.325, 0.014,  0.014] },
    # #   'action':             { shape: (7,),          mean: [0.063, 0.087, -0.090, 0.001, 0.006, -0.005, -0.050],
    # #                                                  std:  [0.336, 0.378,  0.445, 0.039, 0.063,  0.078,  0.999] },
    # # }

    # # Prove that setup() on pretrained policy clobbers pretrained stats with dataset stats.
    # # Pi05.setup() calls _update_preprocessor_stats(stats_dict) unconditionally when
    # # self.model is not None — fine-tuning path always overwrites pretrained normalization.
    # dm = LeRobotDataModule(repo_id="HuggingFaceVLA/libero", num_workers=0, episodes=[0])
    # policy = Pi05("lerobot/pi05_base")

    # print("\n--- BEFORE setup() ---")
    # print("Pretrained stats (action mean):", policy._dataset_stats)
    # # Identity stats — generic base model.
    # # {
    # #   'observation.state': { shape: (32,), mean: [0.0 x32], std: [1.0 x32] },
    # #   'action':            { shape: (32,), mean: [0.0 x32], std: [1.0 x32] },
    # # }

    # trainer = Trainer(max_epochs=1)
    # dm.setup("fit")
    # trainer.datamodule = dm
    # policy._trainer = trainer
    # policy.setup("fit")

    # print("\n--- AFTER setup() with LIBERO ---")
    # print("Post-setup stats (action mean):", policy._dataset_stats)
    # print("\nPretrained normalization overwritten with LIBERO stats.")
    # # Identity stats gone — now LIBERO episode-0 stats.
    # # {
    # #   'observation.image':  { shape: (3, 256, 256), mean: [0.487, 0.440, 0.405], std: [0.202, 0.233, 0.223] },
    # #   'observation.image2': { shape: (3, 256, 256), mean: [0.514, 0.456, 0.421], std: [0.231, 0.264, 0.250] },
    # #   'observation.state':  { shape: (8,),          mean: [-0.047, 0.034, 0.765, 2.972, -0.220, -0.126, 0.027, -0.027],
    # #                                                  std:  [ 0.105, 0.152, 0.379, 0.344,  0.907,  0.325, 0.014,  0.014] },
    # #   'action':             { shape: (7,),          mean: [0.063, 0.087, -0.090, 0.001, 0.006, -0.005, -0.050],
    # #                                                  std:  [0.336, 0.378,  0.445, 0.039, 0.063,  0.078,  0.999] },
    # # }

    # # benchmark.evaluate() does NOT call setup() — it just runs rollouts with whatever
    # # preprocessor stats are already loaded. So the benchmark itself doesn't clobber stats.
    # # The damage is already done above by setup(). The benchmark sees and uses the LIBERO stats,
    # # not the pretrained identity stats — causing normalization mismatch at inference time.
    benchmark = LiberoBenchmark(task_suite="libero_10", num_episodes=20)

    # print("\n--- Stats the benchmark will use (clobbered, NOT pretrained) ---")
    # print("action mean:", policy._dataset_stats["action"]["mean"])
    # print("observation.state mean:", policy._dataset_stats["observation.state"]["mean"])
    # benchmark.evaluate(policy) would run here with wrong normalization
```

### Steps to Reproduce

-

### Expected Behavior

We should add a `_update_preprocessor_stats function` something like:

```python
def _update_preprocessor_stats(
    self,
    dataset_stats: dict[str, dict[str, list[float] | str | tuple]],
) -> None:
    from .preprocessor import make_smolvla_preprocessors
    self._preprocessor, self._postprocessor = make_smolvla_preprocessors(
        max_state_dim=self.config.max_state_dim,
        max_action_dim=self.config.max_action_dim,
        stats=dataset_stats,
        image_resolution=self.config.resize_imgs_with_padding,
        max_token_len=self.config.tokenizer_max_length,
        token_pad_type=self.config.pad_language_to,
    )
```

Then in `setup` we should match Pi05's pattern:

```python
def setup(self, stage: str) -> None:
    del stage

    datamodule = self.trainer.datamodule
    train_dataset = datamodule.train_dataset

    if not isinstance(train_dataset, Dataset):
        raise TypeError(f"Expected physicalai Dataset, got {type(train_dataset)}")

    stats_dict = train_dataset.stats

    if self.model is not None:
        # Fine-tuning path: model weights stay, preprocessors rebuilt with new stats.
        self._update_preprocessor_stats(stats_dict)
        reformat_dataset_to_match_policy(self, datamodule)
        return

    self.hparams["dataset_stats"] = stats_dict
    self._initialize_model(stats_dict)
    reformat_dataset_to_match_policy(self, datamodule)
```

### Actual Behavior

-

### Component

Library (VLA framework)

### Environment

```markdown

```

### Logs / Error Output

```shell

```
